Neos.Flow/Classes/Persistence/Doctrine/DataTypes/JsonArrayType.php

```diff
@@ -93,58 +68,54 @@ public function getMappedDatabaseTypes(AbstractPlatform $platform)
      * @param mixed $value The value to convert.
      * @param AbstractPlatform $platform The currently used database platform.
      * @return array The PHP representation of the value.
+     * @throws ConversionException
+     * @throws TypeConverterException
      */
-    public function convertToPHPValue($value, AbstractPlatform $platform)
+    public function convertToPHPValue($value, AbstractPlatform $platform): array
     {
-        $this->initializeDependencies();
+        $value = parent::convertToPHPValue($value, $platform);
 
-        switch ($platform->getName()) {
-            case 'postgresql':
-                $value = (is_resource($value)) ? stream_get_contents($value) : $value;
-                $array = parent::convertToPHPValue($value, $platform);
-                break;
-            default:
-                $array = parent::convertToPHPValue($value, $platform);
-        }
-        if (is_array($array)) {
-            $this->decodeObjectReferences($array);
+        if (!is_array($value)) {
+            throw new \InvalidArgumentException(sprintf('The JsonArrayType only converts arrays, %s given', gettype($value)), 1663056939);
         }
 
-        return $array;
+        $this->initializeDependencies();
+        $this->decodeObjectReferences($value);
+
+        return $value;
     }
 
     /**
      * Converts a value from its PHP representation to its database representation
      * of this type.
      *
-     * @param mixed $array The value to convert.
+     * @param mixed $value The value to convert.
      * @param AbstractPlatform $platform The currently used database platform.
-     * @return mixed The database representation of the value.
+     * @return string|bool|null The database representation of the value.
+     * @throws \JsonException
      */
-    public function convertToDatabaseValue($array, AbstractPlatform $platform)
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): string|bool|null
```

Comment:
```suggestion
    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
```
since we use now JSON_THROW_ON_ERROR 🎉